FaustCode/owl.cpp

```diff
@@ -347,6 +347,31 @@ class OwlButton : public OwlParameterBase {
   }
 };
 
+class OwlCheckbox : public OwlParameterBase {
+protected:
+    PatchButtonId fButton; // OWL button id : PUSHBUTTON, ...
+    bool wasHigh = false; // Flag for edge detection
+    bool state = false;   // Current state
+public:
+    OwlCheckbox(Patch* pp, PatchButtonId button, FAUSTFLOAT* z, const char* l)
+        : OwlParameterBase(pp, z, false)
+        , fButton(button) {}
+    void update() {
+        bool isHigh = fPatch->isButtonPressed(fButton);
+        if (isHigh  && !wasHigh) {
+            // Rising edge detected
+            state = !state;
+            wasHigh = true;
+        }
+        else if (!isHigh && wasHigh) {
+            // Falling edge detected
+            wasHigh = false;
+        }
```

Comment:
I will double check, but I think it may not work if this code is not running in callback for button state change like your C++ code. If this runs for every audio block, it would reset button state while it's held high.

Reply:
Why not get rid of both conditionals here instead ;-)